compiler/semmagic.nim

```diff
@@ -197,6 +197,8 @@ proc evalTypeTrait(c: PContext; traitCall: PNode, operand: PType, context: PSym)
     let complexObj = containsGarbageCollectedRef(t) or
                      hasDestructor(t)
     result = newIntNodeT(toInt128(ord(not complexObj)), traitCall, c.idgen, c.graph)
+  of "hasDefault":
+    result = newIntNodeT(toInt128(ord(not operand.requiresInit)), traitCall, c.idgen, c.graph)
```

Comment:
Could it be called `hasRequiresInit` something? A typetrait `hasDefault` might be used for types with default field names.

Reply:
`hasDefault(Foo) == true` as it should be, isn't it ?

Reply:
But It might be expected

```nim
type
  Foo = object
     id: int
doAssert not hasDefault(Foo)
```

Reply:
We could call such function `hasDefaultField`.

Reply:
Okay